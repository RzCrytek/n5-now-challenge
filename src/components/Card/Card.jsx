import PropTypes from 'prop-types';

import { CardInfo, CardName, CardPicture, CardSummary, CardWrapper } from './card.styled';

import imgNoPhoto from '@/assets/images/no-photo.png';

const Card = ({ character }) => {
  return (
    <CardWrapper>
      <CardPicture>
        <img src={character.image || imgNoPhoto} alt="" />
      </CardPicture>
      <CardSummary>
        <CardName>{character.name}</CardName>
        <CardInfo>
          <p>
            <span>
              <strong>GÃ©nero: </strong>
              {character.gender}
            </span>

            <span> | </span>

            <span>
              <strong>Especie: </strong>
              {character.species}
            </span>
          </p>
        </CardInfo>
      </CardSummary>
    </CardWrapper>
  );
};

Card.propTypes = {
  character: PropTypes.object.isRequired,
};

export default Card;
