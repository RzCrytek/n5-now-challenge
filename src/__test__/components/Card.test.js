import { render } from '@testing-library/react';
import Card from '../../components/card/Card';

describe('componente <Card>', () => {
  const character = {
    name: 'Rick Sanchez',
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    gender: 'Male',
    species: 'Human',
  };

  test('match con el snapshot', () => {
    const { container } = render(<Card character={character} />);
    expect(container).toMatchSnapshot();
  });
});
