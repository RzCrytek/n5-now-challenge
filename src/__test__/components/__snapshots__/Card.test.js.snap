// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente <Card> match con el snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ kaHELt"
  >
    <picture
      class="sc-gsnTZi cmTymU"
    >
      <img
        alt=""
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
    </picture>
    <div
      class="sc-dkzDqf bBcZg"
    >
      <h2
        class="sc-hKMtZM bxdDSM"
      >
        Rick Sanchez
      </h2>
      <div
        class="sc-eCYdqJ hDIDHp"
      >
        <p>
          <span>
            <strong>
              GÃ©nero: 
            </strong>
            Male
          </span>
          <span>
             | 
          </span>
          <span>
            <strong>
              Especie: 
            </strong>
            Human
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
